src/pipeline-provider.ts

```diff
@@ -29,7 +38,12 @@ class PipelineSlot {
 * Class responsible for creating the navigation pipeline.
 */
 export class PipelineProvider {
+
   static inject() { return [Container]; }
+  /**@internal */
+  container: Container;
+  /**@internal */
+  steps: (Function | PipelineSlot)[];
```

Comment:
I don't think steps is ever a function.

Reply:
In the constructor of this class, we are adding some step constructors for later instantiation. like this [line](https://github.com/aurelia/router/pull/615/files/f41a488866ce19b1e3e2bc859262665818990519#diff-e0658779b23fb0d91b6c56ce52c62769R53)

Reply:
They are all PipelineSteps. The placeholders are also PipelineSteps.